### PR TITLE
Mock out HTTP requests in data tests

### LINKED ISSUE #371: Test relies on internet connection, and access live site
I ran the foia/contacts/ tests without an internet connection, and the following test failed. Unit tests should not rely on accessing a webpage. 

```
(foia-hub)vagrant@precise64:/vagrant/code/foia/contacts$ nosetests
............................E......................
======================================================================
ERROR: Verify that the correct years are retrieved
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vagrant/code/foia/contacts/tests/tests.py", line 687, in test_get_years
    years = processing_time_scraper.get_years()
  File "/vagrant/code/foia/contacts/processing_time_scraper.py", line 204, in get_years
    r = requests.get('http://www.foia.gov/data.html')
  File "/home/vagrant/.virtualenvs/foia-hub/lib/python3.4/site-packages/requests/api.py", line 60, in get
    return request('get', url, **kwargs)
  File "/home/vagrant/.virtualenvs/foia-hub/lib/python3.4/site-packages/requests/api.py", line 49, in request
    return session.request(method=method, url=url, **kwargs)
  File "/home/vagrant/.virtualenvs/foia-hub/lib/python3.4/site-packages/requests/sessions.py", line 457, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/vagrant/.virtualenvs/foia-hub/lib/python3.4/site-packages/requests/sessions.py", line 569, in send
    r = adapter.send(request, **kwargs)
  File "/home/vagrant/.virtualenvs/foia-hub/lib/python3.4/site-packages/requests/adapters.py", line 407, in send
    raise ConnectionError(err, request=request)
nose.proxy.ConnectionError: ('Connection aborted.', gaierror(-3, 'Temporary failure in name resolution'))
-------------------- >> begin captured logging << --------------------
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): www.foia.gov
--------------------- >> end captured logging << ---------------------
```

### PR DESCRIPTION
**Pull request by konklone**
This uses [`vcrpy`](https://github.com/kevin1024/vcrpy) to mock out HTTP requests as YAML files in test fixtures. To regenerate cached HTML, delete the relevant YAML file, re-run the tests (with a network connection), and then commit the changes.

The tests can now be run without a network connection.

This involved some refactoring to isolate the network calls, so that they could be mocked inside the tests. `vcrpy` does not do universal monkeypatching on `requests` or `urllib`, so the request call has to be made inside the vcr block. I think the general modularization of parsing from downloading is good for the scraper, though I don't think passing in optional arguments like I am is the ideal solution. Nonetheless, I think it's the most effective way of getting this done without extensive refactoring.

Fixes https://github.com/18F/foia-hub/issues/371.

### CODE REVIEW

File: .python-version [hunk 1]
```diff
@@ -0,0 +1 @@
+3.4.2
```
Comment (khandelwal) [inline .python-version:1]: Oh, hey are we changing this version across the board? Does that mean I have to change something in my pyenv to make this work?

File: contacts/README.md [hunk 2]
```diff
@@ -183,3 +183,9 @@ with:
 ```bash
 nosetests
 ```
+
+To run an individual test file, e.g. `tests/tests.py`:
+
+```bash
+python -m unittest discover -s tests -p tests.py
```
Comment (khandelwal) [inline contacts/README.md:190]: Is there a specific reason we're no longer using nosetests? I'm curious.

File: contacts/scraper.py [hunk 3]
```diff
@@ -234,7 +234,9 @@ def read_manual_data(agency_abbr, manual_data_dir='manual_data'):
     if os.path.isdir(manual_data_dir):
         filename = agency_yaml_filename(manual_data_dir, agency_abbr)
         if os.path.exists(filename):
-            manual_data = yaml.load(open(filename, 'r'))
+            f = open(filename, 'r')
```
Comment (khandelwal) [inline contacts/scraper.py]: You can replace the open(), close() with a 'with open(filname, 'r') statement, and that might be a little cleaner.

File: contacts/tests/tests.py [hunk 4]
```diff
@@ -12,6 +12,13 @@
 
 import processing_time_scraper
 
+# HTTP requests are mocked out with vcrpy and requests
+import vcr
+import requests
+my_vcr = vcr.VCR(
```
Comment (geramirez) [inline contacts/tests/tests.py]: New tests are great :+1: They work for me, but I think some of the change are missing PEP8. I can run flake8 over the changed files and make the changes if you want.

File: .python-version (hunk 1 above)
Comment (konklone) [inline .python-version:1]: There was no `.python-version` in this repository, so I figured it'd be fine to define one. I didn't realize anyone else but me was using pyenv -- all you'd need to do is `pyenv install 3.4.2` if you want this to work.

File: contacts/README.md (hunk 2 above)
Comment (konklone) [inline contacts/README.md:190]: We're still using it, which is why I didn't delete the lines above. But it's valuable to be able to run a specific test file, _and_ to run it without the `nosetests` wrapper. `nosetests` apparently suppresses all kinds of warnings and debug output, which now prints out when running with `python` directly. I fixed a couple of those in this PR, by closing file handles, but there are a few others that are more inscrutable (though they do not look like active bugs, just deprecation notices caused by our dependencies).

File: contacts/tests/tests.py (hunk 4 above)
Comment (konklone) [inline contacts/tests/tests.py]: That would be wonderful. And would you mind addressing @khandelwal's suggestion above to use the `with` block to handle file opening?

Comment (geramirez) [inline contacts/tests/tests.py]: Sure, no problem.

**Comment by geramirez** [comment]
Just made the final updates.

**Comment by konklone** [comment]
@khandelwal, looking to you for merge :hammer:.

**Comment by khandelwal** [comment]
:hammer:  deployed

**Pull request merged**